### DISCUSSION TITLE
Data Egress

### DISCUSSION THREAD

**Discussion by paulmccrady** [Azure/PSRule.Rules.Azure#2481] [Q&A] [answered]
Does PSRule or any of the other modules egress data to Microsoft or any other external destination ?

**Comment by BernieWhite** [comment, marked as answer]
Hi @paulmccrady. We can't talk to other tools that customers might create or future versions. However as of the current version of PSRule for Azure v1.30.0 and PSRule v2.9.0:

- We don't collect telemetry, PowerShell does by default, but this can be opt-ed out. https://azure.github.io/PSRule.Rules.Azure/faq/#collection-of-telemetry
- You may choose to configure: https://github.com/microsoft/PSRule.Monitor.
- By default, summary within GitHub Actions and Azure Pipelines integrations is enabled (if you choose to use them) which will attach information to the pipeline output hosted in GitHub or Azure DevOps. See:
  - https://github.com/microsoft/ps-rule
  - https://github.com/microsoft/PSRule-pipelines/blob/main/docs/tasks.md#ps-rule-assert

Does this answer the question?

**Reply by paulmccrady** [reply]
It does yes thank you.
